From 4ef28773cc248e2ef2933de14ea599b8418c5d96 Mon Sep 17 00:00:00 2001
From: David Li <li.davidm96@gmail.com>
Date: Fri, 14 Mar 2025 12:45:59 +0900
Subject: [PATCH 1/4] cross_compile_grpc_cpp_plugin_path

---
 cmake/opentelemetry-proto.cmake | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/cmake/opentelemetry-proto.cmake b/cmake/opentelemetry-proto.cmake
index c77a874..188b793 100644
--- a/cmake/opentelemetry-proto.cmake
+++ b/cmake/opentelemetry-proto.cmake
@@ -230,6 +230,15 @@ set(PROTOBUF_GENERATED_FILES
 set(PROTOBUF_GENERATE_DEPENDS ${PROTOBUF_PROTOC_EXECUTABLE})
 
 if(WITH_OTLP_GRPC)
+  # When cross-compiling, look for the gRPC plugin
+  # next to the Protobuf compiler
+  get_filename_component(PROTOC_PATH ${PROTOBUF_PROTOC_EXECUTABLE} DIRECTORY)
+  find_program(gRPC_CPP_PLUGIN_EXECUTABLE grpc_cpp_plugin
+               HINTS ${PROTOC_PATH}
+               NO_CMAKE_PATH
+               NO_CMAKE_ENVIRONMENT_PATH)
+  message(STATUS "gRPC_CPP_PLUGIN_EXECUTABLE=${gRPC_CPP_PLUGIN_EXECUTABLE}")
+
   list(APPEND PROTOBUF_GENERATE_DEPENDS ${gRPC_CPP_PLUGIN_EXECUTABLE})
   list(APPEND PROTOBUF_COMMON_FLAGS
        "--grpc_out=generate_mock_code=true:${GENERATED_PROTOBUF_PATH}"
