From 3ff5c9e5476d3ae18da112872a2f4c6aa7941ebc Mon Sep 17 00:00:00 2001
From: David Li <li.davidm96@gmail.com>
Date: Fri, 14 Mar 2025 12:45:59 +0900
Subject: [PATCH 1/4] cross_compile_grpc_cpp_plugin_path

---
 cmake/opentelemetry-proto.cmake | 17 +++++++----------
 1 file changed, 7 insertions(+), 10 deletions(-)

diff --git a/cmake/opentelemetry-proto.cmake b/cmake/opentelemetry-proto.cmake
index 06d6f76..ed5e3a8 100644
--- a/cmake/opentelemetry-proto.cmake
+++ b/cmake/opentelemetry-proto.cmake
@@ -198,16 +198,13 @@ foreach(IMPORT_DIR ${PROTOBUF_IMPORT_DIRS})
 endforeach()
 
 if(WITH_OTLP_GRPC)
-  if(CMAKE_CROSSCOMPILING)
-    find_program(gRPC_CPP_PLUGIN_EXECUTABLE grpc_cpp_plugin)
-  else()
-    if(TARGET gRPC::grpc_cpp_plugin)
-      project_build_tools_get_imported_location(gRPC_CPP_PLUGIN_EXECUTABLE
-                                                gRPC::grpc_cpp_plugin)
-    else()
-      find_program(gRPC_CPP_PLUGIN_EXECUTABLE grpc_cpp_plugin)
-    endif()
-  endif()
+  # When cross-compiling, look for the gRPC plugin
+  # next to the Protobuf compiler
+  get_filename_component(PROTOC_PATH ${PROTOBUF_PROTOC_EXECUTABLE} DIRECTORY)
+  find_program(gRPC_CPP_PLUGIN_EXECUTABLE grpc_cpp_plugin
+               HINTS ${PROTOC_PATH}
+               NO_CMAKE_PATH
+               NO_CMAKE_ENVIRONMENT_PATH)
   message(STATUS "gRPC_CPP_PLUGIN_EXECUTABLE=${gRPC_CPP_PLUGIN_EXECUTABLE}")
 endif()
 
